File: redsheet-portal/src/app/database-models/stakeholder-type.ts
import { required, maxLength, range, nested } from '@rx/annotations';
import { ProjectStakeholder,  } from './'
export class StakeholderType {
    constructor(stakeholderType?: StakeholderType )  {
        let properties = [ "actionRequried", "className", "stakeholderTypeId", "stakeholderTypeName", "projectStakeholders",];
        for (let property of properties)
            if (stakeholderType && stakeholderType[property])
                this[property] = stakeholderType[property];
    }
 
    @required()
    @maxLength(200)
	actionRequried : string =   undefined;
 
    @required()
    @maxLength(200)
	className : string =   undefined;
 
	stakeholderTypeId : number =   0 ;
 
    @required()
    @maxLength(200)
	stakeholderTypeName : string =   undefined;
	@nested(ProjectStakeholder)
	projectStakeholders: ProjectStakeholder[];



}

File: RedSheet.DbEntities/Models/StakeholderType.cs

using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations.Schema;
using System.ComponentModel.DataAnnotations;
using Microsoft.EntityFrameworkCore;
using Rx.Core.Data.Attributes;
using RedSheet.DbEntities.Enums;

namespace RedSheet.DbEntities.Models
{
	[Table("StakeholderTypes", Schema = "dbo")]
	[Serializable]
	public partial class StakeholderType  
	{
 
	[Required] 
	[MaxLength(200)]
	public string ActionRequried { get; set; } 
 
	[Required] 
	[MaxLength(200)]
	public string ClassName { get; set; } 
 
	[DatabaseGenerated(DatabaseGeneratedOption.Identity)] 
	[Key]
	public int StakeholderTypeId { get; set; } 
 
	[Required] 
	[MaxLength(200)]
	public string StakeholderTypeName { get; set; } 
	[InverseProperty("StakeholderType")]
	public ICollection<ProjectStakeholder> ProjectStakeholders { get; set; } 


	public StakeholderType()
	{
	this.ProjectStakeholders = new HashSet<ProjectStakeholder>();

    }

    }
}

